### ISSUE TITLE
[Feature] Add support for Fastly Rate Limit Variables

### ISSUE THREAD

**Issue by davinci26** [ysugimoto/falco#59]
Hey,

We are interested in contributing support for https://developer.fastly.com/reference/vcl/declarations/ratecounter and all the other rate-limit variables that fastly offers.

**Comment by ysugimoto** [comment]
Sounds nice, I wasn't aware of that syntax.

Fortunately, falco already could lint for ratecounter-related functions and properties like `ratecounter.XXXX.bucket.10s`.
To get this feature, we need to add parser and linter function  of:

```vcl
// I think this is declared in root scope
ratecounter requests_rate {
}
```

It might need to improve core logic e.g AST definition but I really glad if you could do this 😃

**Comment by davinci26** [comment]
Yeah that is true, this needs to go on the root scope, really good point.

> Fortunately, falco already could lint for ratecounter-related functions and properties like ratecounter.XXXX.bucket.10s.

Honestly I havent checked much about this, just used falco in our VCL corpus and it had a problem with ratelimit stuff but I didnt focus on which part.

**Comment by ysugimoto** [comment]
Got it.  we can simply implement AST and context I think. I'll do that.

**Comment by shadialtarsha** [comment]
I am preparing a PR for adding AST declaration and will raise it today (hopefully)

**Comment by ysugimoto** [comment]
Resolved by #60
